src/ci/github-actions/jobs.yml

```diff
@@ -212,11 +212,62 @@ auto:
   - image: dist-x86_64-netbsd
     <<: *job-linux-4c
 
-  - image: i686-gnu
-    <<: *job-linux-8c
+  # The i686-gnu job is split into multiple jobs to run tests in parallel.
+  #
+  # The `src/*` and `tests/*` tests are skipped to speed up this slow job.
+  # These tests can be skipped because they are unlikely to be platform specific.
+  #
+  # The other tests are skipped because they run in i686-gnu-2.
+  - image: i686-gnu-1
+    env:
+      IMAGE: i686-gnu
+      SCRIPT: >-
+        python3 ../x.py --stage 2 test
+        --skip compiler
+        --skip src
+        --skip tests/rustdoc-js
+    <<: *job-linux-4c
 
-  - image: i686-gnu-nopt
-    <<: *job-linux-8c
+  # Skip tests that run in i686-gnu-1
+  - image: i686-gnu-2
+    env:
+      IMAGE: i686-gnu
+      SCRIPT: >-
+        python3 ../x.py --stage 2 test
+        --skip tests
+        --skip coverage-map
+        --skip coverage-run
+        --skip library
+        --skip tidyselftest
+        --skip src/bootstrap
+        --skip src/tools/error_index_generator
+        --skip src/tools/linkchecker
+    <<: *job-linux-4c
+
+  # The i686-gnu-nopt job is split into multiple jobs to run tests in parallel.
+  # i686-gnu-nopt-1 skips tests that run in i686-gnu-nopt-2
+  - image: i686-gnu-nopt-1
+    env:
+      IMAGE: i686-gnu-nopt
+      SCRIPT: >-
+        python3 ../x.py --stage 2 test
+        --skip compiler
+        --skip src
+    <<: *job-linux-4c
+
+  # Skip tests that run in i686-gnu-nopt-1
+  - image: i686-gnu-nopt-2
+    env:
+      IMAGE: i686-gnu-nopt
+      SCRIPT: >-
+        python3 ../x.py test --stage 0 --config /config/nopt-std-config.toml library/std &&
+        python3 ../x.py --stage 2 test
+        --skip tests
+        --skip coverage-map
+        --skip coverage-run
+        --skip library
+        --skip tidyselftest
```

Comment:
we could extract the duplication of `--skip` by using yml anchors, what do you think?

Reply:
Trying to use anchors sounds good, if it can be somehow interpolated into the rest of the command.

Reply:
I extracted everything exept `i686-gnu-nopt-2`. I still need to run `@bors try` after these changes but how does it look to you?
I think it should be fine leaving `i686-gnu-nopt-2` like this.

We could extract `i686-gnu-nopt-2` skips by using env variables, but it could complicate the yaml.

Reply:
Let's leave it like this.